urbackupserver/www2/src/features/backups/BackupContentTable.tsx

```diff
@@ -0,0 +1,283 @@
+import { format_size, formatDatetime } from "../../utils/format";
+import { urbackupServer } from "../../App";
+import { Breadcrumbs } from "../../components/Breadcrumbs";
+import { makeBackupsBreadcrumbs } from "./makeBackupsBreadcrumbs";
+import { TableWrapper } from "../../components/TableWrapper";
+
+const useStyles = makeStyles({
+  heading: {
+    display: "flex",
+    gap: tokens.spacingHorizontalM,
+    // Negate the margin top and left values
+    // to keep breadcrumbs aligned to initial top-left position
+    marginInlineStart: "-7px",
+    marginBlockStart: "-7px",
+  },
+  headingAction: {
+    marginInlineStart: "auto",
+  },
+});
+
+const tableStyles = {
+  name: {
+    display: "flex",
+    gap: tokens.spacingHorizontalS,
+    alignItems: "center",
+  },
+  file: {
+    color: tokens.colorBrandBackground,
+  },
+};
+
+export const columns: TableColumnDefinition<File>[] = [
+  createTableColumn<File>({
+    columnId: "name",
+    renderHeaderCell: () => {
+      return "Name";
+    },
+    renderCell: (item) => {
+      return (
+        <TableCellLayout>
+          <span style={tableStyles.name}>
+            {item.dir ? (
+              <Folder20Filled />
+            ) : (
+              <Document20Filled style={tableStyles.file} />
+            )}
+            {item.name}
+          </span>
+        </TableCellLayout>
+      );
+    },
+  }),
+  createTableColumn<File>({
+    columnId: "size",
+    renderHeaderCell: () => {
+      return "Size";
+    },
+    renderCell: (item) => {
+      return (
+        <TableCellLayout>
+          {item.size ? format_size(item.size) : ""}
+        </TableCellLayout>
+      );
+    },
+  }),
+  createTableColumn<File>({
+    columnId: "created",
+    renderHeaderCell: () => {
+      return "Created";
+    },
+    renderCell: (item) => {
+      if (!item.creat) {
+        return "-";
+      }
+
+      return <TableCellLayout>{formatDatetime(item.creat)}</TableCellLayout>;
+    },
+  }),
+  createTableColumn<File>({
+    columnId: "mod",
+    renderHeaderCell: () => {
+      return "Last modified";
+    },
+    renderCell: (item) => {
+      if (!item.mod) {
+        return "-";
+      }
+
+      return <TableCellLayout>{formatDatetime(item.mod)}</TableCellLayout>;
+    },
+  }),
+  createTableColumn<File>({
+    columnId: "access",
+    renderHeaderCell: () => {
+      return "Last accessed";
+    },
+    renderCell: (item) => {
+      if (!item.access) {
+        return "-";
+      }
+
+      return <TableCellLayout>{formatDatetime(item.access)}</TableCellLayout>;
+    },
+  }),
+  createTableColumn<File>({
+    columnId: "actions",
+    renderHeaderCell: () => "",
+    renderCell: () => {
+      return <Button>List</Button>;
+    },
+  }),
+];
+
+const noneColumns = ["name", "size", "created", "mod", "access"];
+
+export function BackupContentTable() {
+  const { clientId, backupId } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
```

Comment:
Perhaps better to use a class with specific (typed) members for this (depending on what you planned with searchParams in the next steps)

Reply:
I haven't seen these (e.g. [`useSearchParams`](https://reactrouter.com/6.28.0/hooks/use-search-params)) react-router hooks allowing typing (I guess since they're based on web APIs).

Please let me know if I misunderstood the suggestion.

Reply:
Thanks for linking to the documentation. I was thinking it was just for keeping some state, but yeah makes sense if it is actually modifying the location URL.